### PR TITLE
.gitignore enriched with .vscode and .vs folders

### PR DESCRIPTION
**Pull request by frengooo**
I noticed that `.vscode` and `.vs` folders were missing into the .gitignore file.
Usually, I use [gitignore.io](https://www.toptal.com/developers/gitignore) to generate gitignore files.
I generated a new .gitignore file with the following technologies:
 - CSharp
 - VisualStudioCode
 - VisualStudio
 - Rider (C# IDE from JetBrains)

but maybe is so long-winded, you can take a look [here](https://www.toptal.com/developers/gitignore/api/csharp,visualstudio,visualstudiocode,rider) and choose if we can use it or not.

Moreover, can I add the standard `ProjectName.Test` folder for tests or there are reasons for the lack?

### CODE REVIEW

**Review by fredmaggiowski — APPROVED** [review]
Hi @silversoul93 thanks for your contribution!

We can surely merge this gitignore file!

As for the test module: yes you can surely add it and create a minimum set of tests for the template. The fact that is missing is actually a problem so it's more than welcome!

**Comment by frengooo** [comment]
> Hi @silversoul93 thanks for your contribution!
> 
> We can surely merge this gitignore file!
> 
> As for the test module: yes you can surely add it and create a minimum set of tests for the template. The fact that is missing is actually a problem so it's more than welcome!

Hi @fredmaggiowski, thanks for your reply.
Ok, I'll add the test module too.

What about the gitignore.io generated gitignore file? Can we use it or it's too long?
You can see the file [here](https://www.toptal.com/developers/gitignore/api/csharp,visualstudio,visualstudiocode,rider)

File: src/DotNetCore_Template.Tests/TemplateTest.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        {
+        }
+
+        [Test]
+        public void Test1()
+        {
+            Assert.Pass();
+        }
+    }
+}
```
Comment (fredmaggiowski) [inline src/DotNetCore_Template.Tests/TemplateTest.cs]: Use newline at the end of a file

```suggestion
}

```

**Comment by frengooo** [comment]
@fredmaggiowski I created a Test project too and I added it to the solution. Let me know if I have to add the gitignore file generated by gitignore.io too.

**Comment by fredmaggiowski** [comment]
> > Hi @silversoul93 thanks for your contribution!
> > We can surely merge this gitignore file!
> > As for the test module: yes you can surely add it and create a minimum set of tests for the template. The fact that is missing is actually a problem so it's more than welcome!
> 
> Hi @fredmaggiowski, thanks for your reply.
> Ok, I'll add the test module too.
> 
> What about the gitignore.io generated gitignore file? Can we use it or it's too long?
> You can see the file [here](https://www.toptal.com/developers/gitignore/api/csharp,visualstudio,visualstudiocode,rider)

I honestly do not have any strong opinion about the gitignore; I think the one currently in the PR should be fine for most of the users.. If you think the generated one could bring in more value and cover cases we're not thinking about let's bring it in :)

**Comment by frengooo** [comment]
> > > Hi @silversoul93 thanks for your contribution!
> > > We can surely merge this gitignore file!
> > > As for the test module: yes you can surely add it and create a minimum set of tests for the template. The fact that is missing is actually a problem so it's more than welcome!
> > 
> > 
> > Hi @fredmaggiowski, thanks for your reply.
> > Ok, I'll add the test module too.
> > What about the gitignore.io generated gitignore file? Can we use it or it's too long?
> > You can see the file [here](https://www.toptal.com/developers/gitignore/api/csharp,visualstudio,visualstudiocode,rider)
> 
> I honestly do not have any strong opinion about the gitignore; I think the one currently in the PR should be fine for most of the users.. If you think the generated one could bring in more value and cover cases we're not thinking about let's bring it in :)

I don't know, I just use it because allows you to generate a .gitignore file in a fast way.
Let's keep this one for now 😄

**Review by davidebianchi — APPROVED** [review]
Good job, thanks @silversoul93 ! 

About the gitignore, I suggest the use of the global gitignore to exclude custom IDE configuration. But we could use this for now and add field to request.

**Pull request merged**